stingray/io.py

```diff
@@ -92,108 +131,218 @@ def _get_additional_data(lctable, additional_columns):
-
+def _get_detector_id(lctable):
+    """Multi-mission detector id finder
+
+    Examples
+    --------
+    >>> from astropy.io import fits
+    >>> import numpy as np
+    >>> a = fits.Column(name='CCDNR', array=np.array([1, 2]), format='K')
+    >>> t = fits.TableHDU.from_columns([a])
+    >>> det_id1 = _get_detector_id(t.data)
+    >>> a = fits.Column(name='pcuid', array=np.array([1, 2]), format='K')
+    >>> t = fits.TableHDU.from_columns([a])
+    >>> det_id2 = _get_detector_id(t.data)
+    >>> np.all(det_id1 == det_id2)
+    True
+    >>> a = fits.Column(name='asdfasdf', array=np.array([1, 2]), format='K')
+    >>> t = fits.TableHDU.from_columns([a])
+    >>> _get_detector_id(t.data) is None
+    True
+    """
+    for column in ["CCDNR", "ccd_id", "PCUID"]:  # XMM  # Chandra  # XTE
+        for name in lctable.columns.names:
+            if column.lower() == name.lower():
+                return np.array(lctable.field(name), dtype=np.int)
+
+    return None
+
+
+def load_events_and_gtis(
+    fits_file,
+    additional_columns=None,
+    gtistring="GTI,GTI0,STDGTI",
+    gti_file=None,
+    hduname="EVENTS",
+    column="TIME",
+):
     """Load event lists and GTIs from one or more files.
 
-    Loads event list from HDU EVENTS of file ``fits_file``, with Good Time
+    Loads event list from HDU EVENTS of file fits_file, with Good Time
     intervals. Optionally, returns additional columns of data from the same
     HDU of the events.
 
     Parameters
     ----------
     fits_file : str
-        The file name and absolute path to the event file.
-
-    return_limits: bool, optional
-        Return the ``TSTART`` and ``TSTOP`` keyword values
 
+    Other parameters
+    ----------------
     additional_columns: list of str, optional
         A list of keys corresponding to the additional columns to extract from
-        the event HDU (ex.: ``['PI', 'X']``)
+        the event HDU (ex.: ['PI', 'X'])
+    gtistring : str
+        Comma-separated list of accepted GTI extensions (default GTI,STDGTI),
+        with or without appended integer number denoting the detector
+    gti_file : str, default None
+        External GTI file
+    hduname : str, default 'EVENTS'
+        Name of the HDU containing the event list
+    return_limits: bool, optional
+        Return the TSTART and TSTOP keyword values
 
     Returns
     -------
+    retvals : Object with the following attributes:
     ev_list : array-like
-        An array of time stamps of events
-
-    gtis: list of the form ``[[gti0_0, gti0_1], [gti1_0, gti1_1], ...]``
-        Good Time Intervals
-
+        Event times in Mission Epoch Time
+    gti_list: [[gti0_0, gti0_1], [gti1_0, gti1_1], ...]
+        GTIs in Mission Epoch Time
     additional_data: dict
-        A dictionary, where each key is the one specified in ``additional_colums``.
+        A dictionary, where each key is the one specified in additional_colums.
```

Comment:
Additional_data is a dictionary. EvtData is an empty class and we assign a number of attributes (listed here), among whose additional_data.

Reply:
Oh, I see, I think I was confused by the formatting. Can you indent all of the attributes of `retvals` by 4 spaces? Because right now, to someone scanning over the "Returns" section, it looks like it returns `retvals` *and* `ev_list` *and* `gti_list` ...

Reply:
Done!